src/abstractdataframe/join.jl

```diff
@@ -290,6 +298,26 @@ function Base.join(df1::AbstractDataFrame,
 
     joiner = DataFrameJoiner(df1, df2, on)
 
+    # Check merge key validity
+    left_invalid = validate[1] ? any(nonunique(joiner.dfl, joiner.left_on)) : false
+    right_invalid = validate[2] ? any(nonunique(joiner.dfr, joiner.right_on)) : false
+
+    if left_invalid && right_invalid
+        first_error_df1 = findfirst(nonunique(joiner.dfl, joiner.left_on))
+        first_error_df2 = findfirst(nonunique(joiner.dfr, joiner.right_on))
+        throw(ArgumentError("Merge key(s) are not unique in either df1 or df2. " *
```

Comment:
Should it rather be `not unique in df1 and df2` as we know that there are duplicates in both `DataFrame`s?